### DISCUSSION TITLE
Large report size for standalone reports

### DISCUSSION THREAD

**Discussion by KobyGam** [datapane/datapane#73] [General]
Hi.
I tried to create a standalone report using `report.save(path=my_path, standalone=True)`, and the generated report has a size of ~5.3MB.  It seems like most of the report's content is constant, and does not depend on the report blocks (even an "empty" report has a size of 5.3MB).
Is there any intention to reduce the report size in standalone mode in the future?

In addition, I noticed that the report takes a long time to load (~30 sec) on standalone computer with no internet. I don't have any experience with HTML, but it seems that removing lines 5-10 of the generated report fixes this problem. Is this the expected behavior?

Thanks!

**Comment by mands** [comment]
Hi there,

Unfortunately due to the design of the system it's not possible to reduce the size of the HTML file much more - as it includes the entire report display code, all libraries, and all files and dataframes that are part of the report. We've made a few changes that should help with the filesize going forwards, and have some other optimisations we plan to introduce.

The other approach is to publish your reports as private reports that you can share securely via the free Datapane server at datapane.com.
